### PR TITLE
[python] expose new encoder parameters as kwargs of brotli.compress

### PR DESCRIPTION
**Pull request by anthrotype**
I added support for the new BrotliParams to the Python extension, as requested in https://github.com/google/brotli/issues/72.
I also modified the bro.py script so that we can control these parameters from the command line.

I have a question. In the encode.h, it's written that the parameter "quality" has to be in the range from 0 to 11. I presume this range is inclusive. Now, if I set the quality between 1 and 11, the output changes accordingly. However, if I set it to 0, the output is always truncated to the same 10 bytes. Does this mean that 0 is an invalid value for the 'quality' parameter?

Thanks.

C.

### CODE REVIEW

File: python/bro.py [hunk 1]
```diff
@@ -2,22 +2,26 @@
...
-__version__ = '0.1'
+__version__ = '1.0'
 
 
-BROTLI_MODES = {
-    'text': brotli.MODE_TEXT,
-    'font': brotli.MODE_FONT
+# default values of encoder parameters
+DEFAULT_PARAMS = {
+    'mode': brotli.MODE_TEXT,
```
Comment (szabadka) [inline python/bro.py]: I just added a MODE_GENERIC compression mode, which is the default. This avoids the confusion that MODE_TEXT could equally mean 'no idea' and 'text for sure' before. Could you change your argument description and default accordingly?

**Comment by anthrotype** [comment]
sure, I'll do that.

File: python/bro.py [hunk 2]
```diff
@@ -49,27 +53,72 @@ def get_binary_stdio(stream):
...
+                        choices=list(range(16, 25)),
+                        help='Base 2 logarithm of the sliding window size. Range is '
+                        '16 to 24. Defaults to 22.')
+    params.add_argument('--lgblock', metavar="LGBLOCK", type=int,
+                        choices=[0] + list(range(16, 25)),
+                        help='Base 2 logarithm of the maximum input block size. '
+                        'Range is 16 to 24. If set to 0, the value will be set based '
+                        'on the quality. Defaults to 0.')
+    above9 = parser.add_argument_group(
+                        'encoder parameters respected only if quality > 9.')
```
Comment (szabadka) [inline python/bro.py]: I think these low-level parameters should not be exposed to the binary. Their purpose is to find a balance between compression speed and compression density, just like the quality parameter. The reason I still have these here is that I don't have a quality 10 yet, which would be a slow but not extremely slow version. Once I have a reasonable compromise there,  I will probably remove these and rely solely on the quality (or maybe the mode as well), just like for qualities < 9.

Comment (anthrotype) [inline python/bro.py]: I see. Which arguments shall I keep or remove precisely?

Comment (szabadka) [inline python/bro.py]: Those under "// These settings will be respected only if quality > 9."
should be removed.

Comment (anthrotype) [inline python/bro.py]: ok

**Comment by anthrotype** [comment]
@szabadka OK, I've rebased on top of master, set MODE_GENERIC as default compression mode, and removed the additional low-level parameters from the command line tool's arguments (but I left them in the module's compress function)

**Comment by szabadka** [comment]
As for quality 0, it is allowed, but currently is the same as quality 1. I found why it is broken now, I will fix it soon.

**Comment by anthrotype** [comment]
thank you!

**Pull request merged**